### PR TITLE
[DOCS] Added docstring example for EpsilonLogitsWarper #24783

### PR DESCRIPTION
**Pull request by sanjeevk-os**
# What does this PR do?

See #24783 
Added docstring example for EpsilonLogitsWarper

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?

## Who can review?

@gante

### CODE REVIEW

**Review by gante — COMMENTED** [review]
In addition to the comments in the lines below, see also [this comment](https://github.com/huggingface/transformers/pull/25361#pullrequestreview-1566967723): let's try to build shorter code for the example :)

File: src/transformers/generation/logits_process.py [hunk 1]
```diff
@@ -490,6 +490,45 @@ class EpsilonLogitsWarper(LogitsWarper):
...
+    >>> # The below sentence is used since the probability of generating `J. Trump` as the next tokens is very high.
+    >>> sentence = "The full name of Donald is Donald"
+
+    >>> # Tokenize the sentence
+    >>> inputs = tokenizer(sentence, return_tensors="pt")
+
+    >>> set_seed(100)
+
+    >>> # We can see that the model generates `J. Trump` as the next token
+    >>> outputs = model.generate(inputs["input_ids"], max_new_tokens=4)
```
[the remark this answers is not in the document]
Comment (sanjeevk-os) [reply src/transformers/generation/logits_process.py:519]: Hi @gante : Can you share some explanation for illustrating the second case you mentioned ? I am trying to understand: How does the case when ordinary multinomial sampling doesn't generate the most common case and applying `epsilon_cutoff` resulting in the generation of common case show that using `epsilon_cutoff` is better at sampling from multiple variety of tokens ? Thanks.

**Comment by sanjeevk-os** [comment]
Hi @gante, addressed your comments.

File: src/transformers/generation/logits_process.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (gante) [reply src/transformers/generation/logits_process.py]: `epsilon_cutoff` is a variation of `top_p`/`top_k`, sharing a common point: it drops some low-probability solutions according to its algorithm. As such, we should be able to show an example where applying it on a sequence that has a low probability token results in a modified sequence :)

**Review by gante — COMMENTED** [review]
The example has 40 lines at the moment, which is too long. Our docs should be concise, otherwise users won't bother reading them :) 

Let's get a case where we can showcase the processor with 2 `generate` calls (one with and another without `epsilon_cutoff`). Note that `set_seed` needs to be called before each `generate` call, otherwise we can't show that `epsilon_cutoff` had an impact (i.e. otherwise the difference can be attributed to sampling, and not to `epsilon_cutoff`)

**Comment by sanjeevk-os** [comment]
> The example has 40 lines at the moment, which is too long. Our docs should be concise, otherwise users won't bother reading them :)
> 
> Let's get a case where we can showcase the processor with 2 `generate` calls (one with and another without `epsilon_cutoff`). Note that `set_seed` needs to be called before each `generate` call, otherwise we can't show that `epsilon_cutoff` had an impact (i.e. otherwise the difference can be attributed to sampling, and not to `epsilon_cutoff`)

Hi @gante made the suggested changes.

**Review by gante — APPROVED** [review]
Thank you for iterating 👍

File: src/transformers/generation/logits_process.py [hunk 2]
```diff
@@ -521,6 +521,34 @@ class EpsilonLogitsWarper(LogitsWarper):
...
+
+    >>> # We can see that the model generates `J. Trump` as the next token
+    >>> outputs = model.generate(**inputs, max_new_tokens=4, do_sample=True)
+    >>> print(tokenizer.decode(outputs[0], skip_special_tokens=True))
+    The full name of Donald is Donald J. Trump –
+
+    >>> set_seed(19)
+    >>> # The use of the `epsilon_cutoff` parameter (best performing values between 3e-4 and 9e-4 from the paper mentioned above) generates tokens
+    >>> # by sampling from a variety of tokens with probabilities greater than or equal to epsilon value. The disadvantage of this sampling is that
+    >>> # if there are many possible tokens to sample from, the epsilon value has to be very small for sampling to occur from all the possible tokens.
```
Comment (ArthurZucker) [inline src/transformers/generation/logits_process.py:545]: These lines are a bit too long should be less than 119 characters

**Review by ArthurZucker — APPROVED** [review]
Left a very small nit! Thanks for working on this! 🤗

File: src/transformers/generation/logits_process.py (hunk 2 above)
Comment (sanjeevk-os) [reply src/transformers/generation/logits_process.py]: @ArthurZucker  : made the suggested change

**Comment by gante** [comment]
@sanjeevk-os Thank you for your contribution 💛

**Pull request merged**